refactor(drivers): replace axios with native fetch API

tylerbutler opened the pull request:
## Summary

Replaces the axios HTTP client library with native `fetch` API across client-side packages to reduce bundle size and simplify dependencies.

## Changes

### Routerlicious Driver (`packages/drivers/routerlicious-driver`)
- Replaced axios with native `fetch` via `cross-fetch` polyfill
- Created `request.cts` with fetch-compatible type definitions (`RequestConfig`, `RequestHeaders`, `RawRequestHeaders`) that maintain API compatibility with server-side axios usage
- Removed `axios.cts` re-export file
- Moved `uuid` from runtime to dev dependencies (only used in tests)

### Driver Utils (`packages/loader/driver-utils`)
- Updated `insecureUrlResolver.ts` to use native `fetch` instead of axios
- Removed axios dependency from package.json

### Azure Client E2E Tests (`packages/service-clients/end-to-end-tests/azure-client`)
- Refactored `AzureClientFactory.ts` to use native `fetch`
- Introduced `FetchResponse` interface to replace `AxiosResponse` type
- Updated response handling with explicit `response.ok` checks (fetch doesn't auto-throw on non-2xx)
- Improved type safety in `getContainerIdFromPayloadResponse` with explicit type guards
- Removed `AxiosResponse` type imports from all test files

tylerbutler committed:
refactor: replace axios with fetch in client libs

tylerbutler committed:
r11s driver

tylerbutler committed:
lockfile

tylerbutler committed:
uuid is only used in tests

tylerbutler commented on packages/drivers/routerlicious-driver/package.json:
No, turns out uuid is used in the tests, so I moved the dep into devdeps. Types were missing before.

tylerbutler committed:
pnpm dedupe

alexvy86 commented on packages/drivers/routerlicious-driver/package.json:
Recent versions of uuid now include type definitions, I think it is indeed unnecessary.

tylerbutler committed:
rm unneeded types

tylerbutler committed:
code review

tylerbutler committed:
rename types

tylerbutler committed:
rm

tylerbutler committed:
lint

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
fix: add explicit error handling for fetch responses

Fetch doesn't automatically throw on non-2xx responses like axios did,
so added explicit response.ok check before parsing JSON to ensure
proper error handling.

alexvy86 commented on packages/drivers/routerlicious-driver/src/request.cts:
I'd actually say we can remove the aliases to the old names, the types are not package-exported so nothing should be using them by their old name.

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
fixes

tylerbutler committed:
feedback

tylerbutler commented on packages/drivers/routerlicious-driver/src/request.cts:
```suggestion
```

tylerbutler committed:
Update packages/drivers/routerlicious-driver/src/request.cts

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

alexvy86 commented on packages/drivers/routerlicious-driver/src/restWrapper.ts:
```suggestion
const buildRequestUrl = (requestConfig: RequestConfig): string =>
```
We'll still need the return type at some point anyway.

alexvy86 commented on packages/service-clients/end-to-end-tests/azure-client/src/test/AzureClientFactory.ts:
```suggestion
				endpoint: "http://localhost:7071",
```
It's really for tinylicious, not sure the new comment adds much. Same on lines 167 and 216 below, I'd remove it.

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

alexvy86 commented on .gitignore:
Nit: I'd revert this as unrelated to the PR.

tylerbutler commented on .gitignore:
```suggestion
.claude/
```

tylerbutler committed:
Update .gitignore

tylerbutler committed:
Merge branch 'main' into bt-rm-axios

tylerbutler committed:
fixes
